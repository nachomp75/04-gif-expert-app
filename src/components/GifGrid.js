import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p className="animate_animated animate__flash">Loading</p>}
      <div className="card-grid animate__animated animate__fadeIn">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
