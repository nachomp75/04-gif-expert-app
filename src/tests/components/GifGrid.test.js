import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid', () => {
  const category = 'Will Smith slap';

  test('should show the component correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images are loaded with useFetchGifs', () => {
    const gifs = [
      {
        id: '123',
        url: 'https://localhost:3000/testing/test1.jpg',
        title: 'Mega test 1',
      },
      {
        id: '456',
        url: 'https://localhost:3000/testing/test2.jpg',
        title: 'Mega test 2',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
