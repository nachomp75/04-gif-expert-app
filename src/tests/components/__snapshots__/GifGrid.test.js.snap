// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid should show items when images are loaded with useFetchGifs 1`] = `
<Fragment>
  <h3>
    Will Smith slap
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="123"
      key="123"
      title="Mega test 1"
      url="https://localhost:3000/testing/test1.jpg"
    />
    <GifGridItem
      id="456"
      key="456"
      title="Mega test 2"
      url="https://localhost:3000/testing/test2.jpg"
    />
  </div>
</Fragment>
`;

exports[`Testing GifGrid should show the component correctly 1`] = `
<Fragment>
  <h3>
    Will Smith slap
  </h3>
  <p
    className="animate_animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;
